pkg/controllerutils/worker.go

```diff
@@ -1,128 +0,0 @@
-// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package controllerutils
-
-import (
-	"context"
-	"fmt"
-	"sync"
-	"time"
-
-	"github.com/go-logr/logr"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/util/workqueue"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-)
-
-// WorkerOptions are options for a controller's worker.
-type WorkerOptions struct {
-	logger logr.Logger
-}
-
-// WorkerOption is a func that mutates WorkerOptions.
-type WorkerOption func(*WorkerOptions)
-
-// WithLogger configures the logr.Logger to use for a controller worker.
-// If set, a logger preconfigured with the name and namespace field will be injected into the context.Context on
-// each Reconcile call. It can be retrieved via log.FromContext.
-func WithLogger(logger logr.Logger) WorkerOption {
-	return func(options *WorkerOptions) {
-		options.logger = logger
-	}
-}
-
-// CreateWorker creates and runs a worker goroutine that just processes items in the
-// specified queue. The worker will run until ctx is cancelled. The worker will be
-// added to the wait group when started and marked done when finished.
-func CreateWorker(ctx context.Context, queue workqueue.RateLimitingInterface, resourceType string, reconciler reconcile.Reconciler, waitGroup *sync.WaitGroup, workerCh chan<- int, opts ...WorkerOption) {
```

Comment:
There are `CreateWorker` in `docs/development/logging.md` - you could clean them up as well.